src/js/node/fs.promises.ts

```diff
@@ -112,7 +112,7 @@ function cp(src, dest, options) {
 }
 
 async function opendir(dir: string, options) {
-  return new (require('node:fs').Dir)(1, dir, options);
+  return new (require("node:fs").Dir)(1, dir, options);
```

Comment:
definition of `require("node:fs").Dir` should be moved up to file scope

Reply:
I didn't write this, I just ran prettier

Reply:
i would have done this but that would cause an infinite recursion. `require` pointing to an internal js file has been setup to be extremely cheap